zokrates_core/lib/wraplibsnarkgadgets.hpp

```diff
@@ -12,11 +12,13 @@ extern "C" {
 #include <stdbool.h>
 #include <stdint.h>
 
-char* _sha256Constraints();
-char* _sha256Witness(const uint8_t* input, int input_length);
+char* _sha256RoundConstraints();
+char* _sha256RoundWitness(const uint8_t* input, int input_length);
 
-char* _shaEth256Constraints();
-char* _shaEth256Witness(const uint8_t* input, int input_length);
+// External interface to free memory 
+void _free_string(char const *str) {
```

Comment:
THis is fine for now but if this is not as clean as we'd want, let's open an issue.